docs/observability: document how to configure metrics scraping (#1732)
openservicemesh/osm: docs/patterns/observability.md

### Automatic bring up

> By default, OSM installation will deploy a Prometheus instance to scrape the sidecar's metrics endpoints. Based on the metrics scraping configuration set by the user, OSM will annotate pods part of the mesh with necessary metrics annotations to have Prometheus reach and scrape the pods to collect relevant metrics. To install Grafana for metrics visualization, set the `enable-grafana` flag to true when installing OSM using the `osm install` command.

Reviewer: ```suggestion
By default, OSM installation will deploy a Prometheus instance to scrape the sidecar's metrics endpoints. Based on the metrics scraping configuration set by the user, OSM will annotate pods that are a part of the mesh with necessary metrics annotations enabling Prometheus to reach and scrape the pods for relevant metrics. To install Grafana for metrics visualization, set the `enable-grafana` flag to true when installing OSM using the `osm install` command.
```

Author: Existing grammar seems fine to me.

      target_label: source_workload_name
```

## Configuring Prometheus metrics scraping

> Metrics scraping can be configured using the `osm metrics` command. By default, OSM does not configure metrics scraping for pods in the mesh. Metrics scraping can be enabled or disabled at namespace scope such that pods belonging to configured namespaces can be enabled or disabled for scraping metrics.

Reviewer: ```suggestion
Metrics scraping can be configured using the `osm metrics` command. By default, OSM **does not** configure metrics scraping for pods in the mesh. Metrics scraping can be enabled or disabled at namespace scope such that pods belonging to configured namespaces can be enabled or disabled for scraping metrics.
```

For metrics to be scraped, the following prerequisites must be met:

> - A Prometheus instance must be deployed either using [automatic bringup of Prometheus](#automatic-bring-up) or using the [bring your own Prometheus guide](#byo-bring-your-own).

Reviewer 2: I'd relax the wording a bit here
```suggestion
- A running service able to scrap Prometheus endpoints. OSM provides configuration for an [automatic bringup of Prometheus](#automatic-bring-up); alternatively users can [bring their own Prometheus](#byo-bring-your-own).
```
